### PR TITLE
Consolidation resiliency fixes in CoreCLR build scripts

### PR DESCRIPTION
**Pull request by trylek**
Backport of my consolidation changes aimed at making principal
CoreCLR build scripts orthogonal to repo consolidation.

Thanks

Tomas

### CODE REVIEW

**Comment by trylek** [comment]
cc: @dotnet/coreclr-infra

**Comment by trylek** [comment]
The error in crossgen2 build is caused by a bug in the init-dotnet script. I'm investigating the proper fix in the runtime repo, I'll update the review once I figure out the proper formulation :-).

File: tests/build.proj [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
   <Target Name="RestorePackage">
     <PropertyGroup>
-      <DotnetCliPath Condition="'$(DotnetCliPath)'==''">$(ProjectDir)..\.dotnet\</DotnetCliPath>
+      <DotnetCliPath Condition="'$(DotnetCliPath)'==''">$(RepoRoot)\.dotnet\</DotnetCliPath>
```
Comment (janvorli) [inline tests/build.proj]: I cannot find where the RepoRoot comes from.

Comment (trylek) [reply tests/build.proj]: This gets set internally in the Arcade scripts.

Comment (janvorli) [reply tests/build.proj]: Ok, I see. I had a two weeks old local checkout which didn't contain that yet.

**Pull request merged**